enumerables.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?
+    selfitem = self
+    if !block_given?
+      selfitem.my_each do |x|
+        return true unless x != false
+      end
+    elsif block_given?
+      selfitem.my_each { |x| return false unless yield(x) }
+    else
+      selfitem.my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?
+    selfitem = self
+    if !block_given?
+      selfitem.my_each do |x|
+        return true if x == true
+      end
+    elsif block_given?
+      selfitem.my_each { |x| return true if yield(x) }
+    else
+      selfitem.my_each { |x| return true if yield(x) }
+    end
+    false
+  end
+
+  def my_none?
+    selfitem = self
+    if !block_given?
+      selfitem.my_each do |x|
+        return false if x == true
+      end
+    elsif block_given?
+      selfitem.my_each { |x| return false if yield(x) }
+    else
+      selfitem.my_each { |x| return false if x }
+    end
+    true
+  end
+
+  def my_count
+    return length unless block_given?
+
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    cont = array.length
+    cont
+  end
```

Comment:
**[REQUIRED]**
* Counts the number of items in enum that are equal to item if an argument is given.
```
Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```